src/index.ts

```diff
@@ -0,0 +1,55 @@
+import * as github from '@actions/github';
+import * as core from '@actions/core';
+import * as semver from 'semver';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { ActionInput, ActionInputParam, VersionDoc, Tag } from './types';
+
+const getTagsPromises = (repositoryNames: string[], inputs: ActionInput) => {
+    const tagsPromises: Promise<Tag>[] = repositoryNames.map(async (repositoryName) => {
+        const { data: tags } = await github.getOctokit(inputs.github_token).rest.repos.listTags({
+            owner: github.context.payload.repository.owner.login,
+            repo: repositoryName,
+        });
+        return tags.filter((tag) => semver.valid(tag.name)).sort((a, b) => semver.rcompare(a.name, b.name))[0];
+    });
+    return tagsPromises;
+};
+
+const getActionInputs = (variables: Array<ActionInputParam>): ActionInput => {
+    return <ActionInput>variables.reduce((obj, variable) => {
+        let value: string = core.getInput(variable.name, variable.options);
+        if (!value) {
+            if (Object.prototype.hasOwnProperty.call(variable, 'default')) {
+                value = variable.default;
+            }
+        }
+        return Object.assign(obj, { [variable.name]: value } );
+    }, {});
+};
+
+async function run(): Promise<void> {
+
+    // Get the GH token and version file path
+    const inputs: ActionInput = getActionInputs([
+        { name: 'github_token', options: { required: true } },
+        { name: 'version_file_path', options: { required: true } },
+    ]);
+
+    // Get the repos to cycle through later
+    const doc: VersionDoc = yaml.load(fs.readFileSync(inputs.version_file_path, 'utf8'));
```

Comment:
A nice to have here might be validation that the yaml has the correct shape. Typescript won't check that the yaml parses to this shape at runtime, so this could mask an issue with bad data. I've used the yup library for this in the past. It has the benfit that if you declare the validation schema to be e.g. `SchemaOf<VersionDoc>` typescript will validate that the the schema has a shape that is valid for the type at compile time.

Reply:
@LiamMacP Re: your test comment  - this is the only test I can think was worth writing but it would still need refactoring into a bit of a silly state, basically a class with a new method with 1 line in it, where 90% of that line is using a library...